### PR TITLE
Add support for edge mode

### PR DESCRIPTION
**Pull request by paulosman**
When run in edge mode, honeyalb and honeyelb will ignore the parent_id segment of Amazon trace headers. This prevents requests from the public internet containing amazon headers from resulting in broken traces (missing root spans).

### CODE REVIEW

File: options/options.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ type Options struct {
 	StateDir        string  `long:"statedir" description:"Directory where ingest state is stored" default:"."`
 	HighAvail       bool    `long:"highavail" description:"Enable high availability ingestion using DynamoDB"`
 	BackfillHr      int     `long:"backfill" description:"The number of hours to increase backfill of log ingestion to with max of 168 hours (1 week)" default:"1"`
+	EdgeMode        bool    `long:"edge_mode" description:"Ignore any parent trace id, if present, from a load balancer"`
```
Comment (MikeGoldsmith) [inline options/options.go:10]: Should we have a default of `false` here?

**Review by MikeGoldsmith — CHANGES_REQUESTED** [review]
LGMT - just one query regarding default value.

File: options/options.go (hunk 1 above)
Comment (paulosman) [reply options/options.go:10]: Good catch. Turns out that boolean flags automatically default to false and can't have an explicit default value! So we're good here.

Comment (MikeGoldsmith) [reply options/options.go:10]: 👍

**Review by MikeGoldsmith — APPROVED** [review]

File: options/options.go (hunk 1 above)
Comment (MikeGoldsmith) [reply options/options.go:10]: Maybe we could add that `false` is the default in the description?

**Pull request merged**